### PR TITLE
MDEP-779 show class names used in undeclared dependencies

### PR DESCRIPTION
**Pull request by josephw**
This adds [the class names tracked in this PR](https://github.com/apache/maven-dependency-analyzer/pull/44) to the UI, when `-Dverbose` is set:

```
[WARNING] Used undeclared dependencies found:
[WARNING]    xml-apis:xml-apis:jar:1.0.b2:compile
[WARNING]       class org.apache.xmlcommons.Version
[WARNING] Unused declared dependencies found:
[WARNING]    dom4j:dom4j:jar:1.6.1:compile
```
It also includes the small changes necessary for any upgrade of `maven-dependency-analyzer` to `1.12`.

To allow this to be reviewed ahead of a release of `maven-dependency-analyzer`, the version is set to `1.12.0-SNAPSHOT`; if approved, it'll presumably be `1.12.0`. As mentioned, it ties the new output to `-Dverbose`; I think this is simpler than adding a new specific property, and less disruptive than enabling it by default.

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MDEP) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MDEP-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MDEP-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
`maven-dependency-analyzer` was released, please rebase your PR.

**Comment by josephw** [comment]
> `maven-dependency-analyzer` was released, please rebase your PR.

Great; I've rebased on top of 7b8ff3d9f9d7d8b.

**Comment by slawekjaranowski** [comment]
We need some tests to confirm regressions in next releases. 
Please add IT test with output verification.

**Comment by slawekjaranowski** [comment]
There is test `src/it/projects/analyze` you can try to extend test verification

File: src/it/projects/mdep-779-analyze-only-verbose-shows-class-names/verify.bsh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.codehaus.plexus.util.*;
+
+String log = FileUtils.fileRead( new File( basedir, "build.log" ) );
```
Comment (slawekjaranowski) [inline src/it/projects/mdep-779-analyze-only-verbose-shows-class-names/verify.bsh:22]: We can use groovy script here ... and we needn't external dependencies.
look for build.log reading in another groovy scripts

File: src/it/projects/mdep-779-analyze-only-verbose-shows-class-names/verify.bsh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+String expected = "[WARNING] Used undeclared dependencies found:\n"
+                + "[WARNING]    xml-apis:xml-apis:jar:1.0.b2:compile\n"
+                + "[WARNING]       class org.apache.xmlcommons.Version\n";
```
Comment (slawekjaranowski) [inline src/it/projects/mdep-779-analyze-only-verbose-shows-class-names/verify.bsh:26]: we must use system native end line ...

**Comment by jira-importer** [comment]
Resolve #1285

**Pull request merged**